### ISSUE TITLE
Migrate `qiskit_nature.properties`

### ISSUE THREAD

**Issue by mrossinek** [Qiskit/qiskit-nature#705]
### What should we add?

As part of #701, all existing modules of Qiskit Nature will be migrated.
This issue outlines the requirements for the existing `properties` module.
This issue has high priority because it blocks further migration of other modules.
This issue is (partially) blocked by #702.

> ⚠️ This migration needs to **copy** the code in order to abide the deprecation process!

Below are the current module contents:
```
qiskit_nature/
    properties/
        Property
        GroupedProperty
        second_quantization/
            DriverMetadata
            SecondQuantizedProperty
            electronic/
                AngularMomentum
                ElectronicDipoleMoment
                ElectronicEnergy
                ElectronicStructureDriverResult
                Magnetization
                ParticleNumber
                bases/
                    ElectronicBasis
                    ElectronicBasisTransform
                integrals/
                    ElectronicIntegrals
                    IntegralProperty
                    OneBodyElectronicIntegrals
                    TwoBodyElectronicIntegrals
            vibrational/
                OccupiedModals
                VibrationalEnergy
                VibrationalStructureDriverResult
                bases/
                    HarmonicBasis
                    VibrationalBasis
                integrals/
                    VibrationalIntegrals
```

Here are the new locations for all of these classes.
```
qiskit_nature/
    second_quantization/
        operator_factories/
            electronic/
                AngularMomentum
                ElectronicDipoleMoment
                ElectronicEnergy
                Magnetization
                ParticleNumber
                bases/
                    ElectronicBasis
                    ElectronicBasisTransform
            vibrational/
                OccupiedModals
                VibrationalEnergy
                bases/
                    HarmonicBasis
                    VibrationalBasis
```

The following classes will **not** be migrated because of they are superseded as indicated:
- the base class `Property` will no longer be needed. There may instead be a new `OperatorFactory`
  base class
- the `GroupedProperty` will also not be needed as the new `Problem` classes will take on the
  responsibility of storing multiple `OperatorFactory` instances
- `DriverMetadata` will no longer be needed as the problem will store its "provenance" information
  (see upcoming issue on the planned HDF5 schema)
- `SecondQuantizedProperty` will not be needed
- `ElectronicStructureDriverResult` will be superseded by the new `ElectronicStructureProblem`
- `VibrationalStructureDriverResult` will be superseded by the new `VibrationalStructureProblem`
- `ElectronicIntegrals` (and its subclasses) will be superseded by `PolynomialTensor` (see #665)
- `VibrationalIntegrals` will be superseded by `PolynomialTensor` (see #665)

**Comment by MarcoBarroca** [comment]
I’m not sure I completely understand but if all that is needed to resolve is to copy the modules around I can do it when #702 is solved and #714 is merged.

**Comment by mrossinek** [comment]
Thank you for the offer!
The `properties` are a bit more involved as there are various API refactorings going on around them at the same time.
Although an initial copy might indeed be straight forward, I will likely handle this particular module myself for time reasons.

I can however suggest that I will send a smaller issue your way once this has progressed a bit further :+1:

**Comment by MarcoBarroca** [comment]
No problem! Just let me know when and how I can help!
